styles/designs/common/pdf/_common-shapes.scss

```diff
@@ -57,3 +57,9 @@
   ),
 ));
 
+@include create_shape('common:::CodeSnippetLineNumbersShape', (
+  _components: (
+    $CodeSnippet__LineNumbers,
+    $CodeSnippet__LineNumbers--Code,
+  ),
+));
```

Comment:
i may have led you astray with the demo snippet saying to set these two components as siblings! in order to use just one `_selector` for this shape, it might be necessary to wrap these as children of one root `CodeSnippetContainer` component. then, `.hljs-ln-numbers` and `.hljs-ln-code` can become `subselector`s on the component.

- [figure](https://github.com/openstax/ce-styles/blob/main/styles/designs/cardboard/parts/_figureimage-shapes.scss#L7) is one example of a container + nested components shape structure. 
- to choose how to set `selector` and `subselector`, just know that the compile process for shapes concatenates the `selector` and one or more `subselector`s, to form a full selector.
- example: caption title label -- [css](https://github.com/openstax/ce-styles/blob/main/styles/output/accounting-pdf.css#L1990) selector is formed from [shape](https://github.com/openstax/ce-styles/blob/main/styles/designs/cardboard/parts/_figureimage-shapes.scss#L12)